packages/block-library/src/utils/get-transformed-attributes.js

```diff
@@ -43,31 +44,19 @@ export function getTransformedAttributes(
 	) {
 		transformedAttributes.ariaLabel = attributes.ariaLabel;
 	}
-	if (
-		hasBlockSupport( newBlockType, 'className' ) &&
-		attributes.className
-	) {
-		transformedAttributes.className = attributes.className;
-	}
```

Comment:
We added this in #72196, but it was redundant because there was already [a hook to convert class names](https://github.com/WordPress/gutenberg/blob/98e9295d7ffe1bc160277ae8f855c0c7ca5ac594/packages/block-editor/src/hooks/custom-class-name.js#L146-L150).